host.go

```diff
@@ -0,0 +1,81 @@
+package p2pgrpc
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-crypto"
+	"github.com/libp2p/go-libp2p-host"
+	"github.com/libp2p/go-libp2p-peer"
+
+	mrand "math/rand"
+
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// MakeBasicHost creates a LibP2P host with a random peer ID listening on the
+// given multiaddress. It will use secio if secio is true.
+func MakeBasicHost(listenPort int, secio bool, randseed int64) (host.Host, error) {
```

Comment:
Not sure if having this part of the library is a good idea. This function is assuming tcp as a default protocol

Reply:
It is assuming many things not only tcp but protocols, key types ..., only used in tests and in the examples folders.